### ISSUE TITLE
Feature Request Summary

### ISSUE THREAD

**Issue by Tereseus** [maxidv88/jammming#7]
After reviewing your feature request your primary goals are to create a login system that will save the access token and hide the searchBar unless credentials are valid or as you put it:

- show the login button and a message to login
- hide the search block
- show an alert if the user try to save the playlist
- If the user has logged in (valid token):
- Save the access token immediately after login to Spotify.
- Store the access token in a persistent storage so after closing the browser, reload the web app or open/close a tab to decrease the number of token requests.
- Show a message with the Spotify’s username and a logout button.

I saw some images of you already implementing the component into your project and as far as I can tell your technical design is totally right. Did you already implement the feature? I cloned your repo but was unable to find the new feature. 

From what your wrote the logic entirely makes sense for this feature and can be implemented. Let me know if you have already implemented it, if not your suggestion is exactly how I would do it also.

**Comment by techno369** [comment]
Yes, it's in the branch unit4. https://github.com/maxidv88/jammming/tree/unit4 . I implemented the described features. Also before that I tried to implement these other features:  1- Showing an audio preview for the tracks. 2- Removing or hiding the tracks from the searchList that were added to the playList and send them back to the search result when the user remove it from playlist (this features needs to be implemented in other way because it doesn't work in a few edge cases when remove the track, also when you play the preview track and add it to the playlist it won't stop because the element is only hidden by css). 3- Use a form tag in the search box to submit pressing the enter key instead of clicking the search button. I know there are some things that can be done better and I wanted to improve or fix them but I was busy.

**Comment by Tereseus** [comment]
Ok sure showing an audio preview you would have to implement a player and see how you could get back the actual track rather than just the information about the track. I believe in Spotify they do not have any type of preview either.

For 2 you kind of answered your own question.

3. Is a feature a lot of people implement you actually implemented much more difficult features which I have to applaud you for.

Do you have any idea how you would do the first?

**Comment by techno369** [comment]
I implemented that 3 features, indeed. In the first one you have a preview track from spotify, in the Track object there is a key called preview_url, it has a string value (A link to a 30 second preview (MP3 format) of the track. null if not available). I used the Spotify.search() function to fetch the preview_track from Spotify (including all the other information, id, song name, etc). I used the html5 tag audio, which was the easies way and set a property to load the tracks when the user click on play button (because by default the player loads all the tracks preview automatically) I did the changes in the track.js file https://github.com/maxidv88/jammming/blob/unit4/src/components/Track/Track.js     https://github.com/maxidv88/jammming/blob/unit4/src/util/Spotify.js

**Comment by techno369** [comment]
Thank you! It was interesting this jamming project.
